File: PDP/DrivingSchoolAppAndroidClient/app/src/main/java/com/example/drivingschoolappandroidclient/api/models/Class.kt
package com.example.drivingschoolappandroidclient.api.models

import com.example.drivingschoolappandroidclient.App
import com.google.gson.annotations.SerializedName
import java.time.LocalTime

data class Class(
    var classId: Int,
    var innerScheduleOfInstructorId: Int,
    var innerScheduleOfInstructor: InnerScheduleOfInstructor,
    var studentId: Int?,
    var student: Student?,
    var startTimeJson: String,
    var durationJson: String,
    var status : Byte
){
    var classStatus : ClassStatus
        get() = ClassStatus.ofValue(status)
        set(value) {status = value.value}
    var startTime: LocalTime
        get() = App.timeFromString(startTimeJson)
        set(value){
            startTimeJson = App.timeToString(value)
        }
    var duration: LocalTime
        get() = App.timeFromString(durationJson)
        set(value){
            durationJson = App.timeToString(value)
        }
    val isOuterClass get() = innerScheduleOfInstructor.IsOuterSchedule;
    val outerScheduleId get() = innerScheduleOfInstructor.outerScheduleId
    val outerScheduleOfInstructor get() = innerScheduleOfInstructor.outerScheduleOfInstructor
    var endTime: LocalTime
        get() = startTime.plusSeconds(duration.toSecondOfDay().toLong())
        set(value) {
            duration = value.minusSeconds(startTime.toSecondOfDay().toLong())
        }
    val startDateTime get() = innerScheduleOfInstructor.dayOfWork.atTime(startTime)
    val endDateTime get() = innerScheduleOfInstructor.dayOfWork.atTime(endTime)
    val date get() = innerScheduleOfInstructor.dayOfWork
    val instructorId get() = instructor.instructorId
    val instructor get() = innerScheduleOfInstructor.instructor
    override fun toString() =
        "$date $startTime-$endTime"

}

File: PDP/DrivingSchoolWebAPI3/DrivingSchoolAPIModels/DbModels/Class.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Globalization;

namespace DrivingSchoolAPIModels
{
    // Занятие
    public class Class
    {
        [Key]
        public int ClassId { get; set; }
        /// <summary>
        /// Тип занятия (внешнее или внутреннее)
        /// </summary>
        public bool IsOuterClass => InnerScheduleOfInstructor.IsOuterSchedule;
        // 
        public int InnerScheduleOfInstructorId { get; set; }
        [ForeignKey(nameof(InnerScheduleOfInstructorId))]
        public InnerScheduleOfInstructor InnerScheduleOfInstructor { get; set; }
        // Если занятие получено из внешнего источника
        public int? OuterScheduleId => InnerScheduleOfInstructor.OuterScheduleId;
        public OuterScheduleOfInstructor? OuterScheduleOfInstructor => InnerScheduleOfInstructor.OuterScheduleOfInstructor;
        // Сам занимающийся ученик
        public int? StudentId { get; set; }
        [ForeignKey(nameof(StudentId))]
        public Student? Student { get; set; }
        // Будет хранить только StartTime, Duration и Status
        public TimeOnly StartTime { get; set; }
        public TimeSpan Duration { get; set; }
        /// <summary>
        /// Проведено ли занятие
        /// </summary>
        [DefaultValue(ClassStatus.Предстоит)]
        public ClassStatus Status { get; set; }
        // Остальные значения будет вычислять в процессе
        [NotMapped]
        public TimeOnly EndTime
        {
            get => TimeOnly.FromTimeSpan(StartTime.ToTimeSpan() + Duration);
            set => Duration = value - StartTime;
        }
        public DateTime StartDateTime => InnerScheduleOfInstructor.DayOfWork.ToDateTime(StartTime);
        public DateTime EndDateTime => InnerScheduleOfInstructor.DayOfWork.ToDateTime(EndTime);
        public DateOnly Date => InnerScheduleOfInstructor.DayOfWork;
        // Инструктор
        public int InstructorId => Instructor.InstructorId;
        public Instructor Instructor => InnerScheduleOfInstructor.Instructor;
        [NotMapped]
        public string StartTimeJson
        {
            get => StartTime.ToString(Consts.TimeOnlyFormatToString);
            set => StartTime = TimeOnly.ParseExact(value, Consts.TimeSpanFormats);
        }
        [NotMapped]
        public string DurationJson
        {
            get => Duration.ToString(Consts.TimeSpanFormatToString);
            set => Duration = TimeSpan.ParseExact(value, Consts.TimeSpanFormats, CultureInfo.CurrentCulture);
        }
    }
}